selinux/services/machined.cil

```diff
@@ -0,0 +1,33 @@
+(type init_exec_t)
+(call system_f (init_exec_t))
+(context init_exec_t (system_u object_r init_exec_t (systemLow systemLow)))
+(filecon "/sbin/init" file init_exec_t)
+(filecon "/sbin/poweroff" any init_exec_t)
+(filecon "/sbin/shutdown" any init_exec_t)
+(filecon "/sbin/dashboard" any init_exec_t)
+
+(type init_t)
+(roletype system_r init_t)
+(allow kernel_t init_exec_t (file (execute)))
+(allow kernel_t init_t (process (transition)))
+(allow init_t init_exec_t (file (execute entrypoint)))
+
+; we create a sealed runc memfd to be shared
+; by all containerd instances and containers
+(type runc_memfd_t)
+(call common_f (runc_memfd_t))
+(allow runc_memfd_t tmpfs_t (filesystem (associate)))
+(allow init_t tmpfs_t (file (relabelfrom)))
+(allow init_t runc_memfd_t (file (relabelto)))
+
+(allow init_t system_service_p (process (transition)))
+; Should not occur unless misconfigured by machined
+(type unconfined_service_t)
+(call system_service_p (unconfined_service_t))
+
+(type sbin_exec_t)
+(call system_f (sbin_exec_t))
+(filecon "/sbin(/.*)?" any (system_u object_r sbin_exec_t (systemLow systemLow)))
+; Typically machined executes LVM, cryptsetup and similar utilities
+; They are short-running, come from the rootfs and do not accept user input, so can be started in init_t domain
+(allow init_t sbin_exec_t (file (execute execute_no_trans)))
```

Comment:
Does machined execute anything apart from sbin things? Does udev execute anything other than modprobe and stuff like `scsi_id`?

Reply:
> Does machined execute anything apart from sbin things?

I think no, also does `kernel_t` has permissions to execute binaries under `/sbin`, I know in azure, kernel actually calls `/sbin/shutdown`

> Does udev execute anything other than modprobe and stuff like scsi_id?

this depends on what's in the udev rules

Reply:
> kernel actually calls /sbin/shutdown

Perhaps I already allowed kernel_t to execute init binary

Reply:
yeh, but this is after switchroot happens :thinking:

Reply:
Will take a look at kernel source

Reply:
`/sbin/poweroff` and `/sbin/reboot` can be called by kernel, yes

Reply:
That might need a typetransition: actually we could avoid transitioning in switchroot but rather make kernel do transition for us